fission/supporttool/resources/crd.go

```diff
@@ -0,0 +1,171 @@
+/*
+Copyright 2018 The Fission Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"log"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/fission/fission"
+	"github.com/fission/fission/controller/client"
+	"github.com/fission/fission/crd"
+)
+
+const (
+	CrdEnvironment = "Environment"
+	CrdFunction    = "Function"
+	CrdPackage     = "Packages"
+
+	CrdHttpTrigger         = "HTTPTrigger"
+	CrdKubeWatcher         = "KubeWatcher"
+	CrdMessageQueueTrigger = "MessageQueue"
+	CrdTimeTrigger         = "TimeTrigger"
+)
+
+type CrdDumper struct {
+	client  *client.Client
+	crdType string
+}
+
+func NewCrdDumper(client *client.Client, crdType string) Resource {
+	return CrdDumper{client: client, crdType: crdType}
+}
+
+func (res CrdDumper) Dump(dumpDir string) {
+
+	switch res.crdType {
+	case CrdEnvironment:
+		items, err := res.client.EnvironmentList(metav1.NamespaceAll)
+		if err != nil {
+			log.Printf("Error getting %v list: %v", res.crdType, err)
+			return
+		}
+
+		for _, item := range items {
+			go func(obj crd.Environment) {
```

Comment:
Does putting this in a goroutine really help?  The API requests might be the slower ones.  Maybe you can just call Dump() itself from a goroutine, since it has no return.